iroh/src/client.rs

```diff
@@ -80,21 +154,21 @@ where
     }
 
     /// List all documents.
-    pub async fn list_docs(&self) -> Result<impl Stream<Item = Result<NamespaceId>>> {
+    pub async fn list(&self) -> Result<impl Stream<Item = Result<NamespaceId>>> {
         let stream = self.rpc.server_streaming(DocListRequest {}).await?;
         Ok(flatten(stream).map_ok(|res| res.id))
     }
 
     /// Get a [`Doc`] client for a single document. Return an error if the document cannot be found.
-    pub async fn get_doc(&self, id: NamespaceId) -> Result<Doc<C>> {
-        match self.try_get_doc(id).await? {
+    pub async fn get(&self, id: NamespaceId) -> Result<Doc<C>> {
```

Comment:
This is a very odd method. The rust convention if there is a `try_foo` and `foo` is to panic if the thing is missing, so I think we should only have `get` and it should return `Result<Option>`

Reply:
My reasoning for a method that returns just `Result<Doc>` is that I would think that in the vast majority of cases you have a document ID and not found equals error, so skipping the `ok_or_else` or `context` call makes application code nicer to read.

No nice naming scheme comes to mind, so I changed it for now to always return `Result<Option<Doc>>`.